primedev/mods/modsavefiles.cpp

```diff
@@ -74,6 +74,24 @@ template <ScriptContext context> void SaveFileManager::SaveFileAsync(fs::path fi
 	std::thread writeThread(
 		[mutex, file, contents]()
 		{
+			// Check if has extension and return early if not
+			if (!file.has_extension())
+			{
+				spdlog::error("SAVE FAILED!");
+				spdlog::error("No file extension specified");
+			}
+
+			// TODO: move into list of global consts?
+			std::set<std::string> whitelist = {".txt", ".json"};
```

Comment:
Is there a reason for this being a `set` instead of like an `vector` or `array` or is it just an arbitrary choice?

Reply:
Mostly arbitrary choice

Reply:
Then it should probably be a vector.

Reply:
can probably also be made `static const`

Reply:
@barnabwhy done via 314b86aead1f3bd2e20840d493178b07113571eb

@catornot going back to this, wouldn't a set make more sense as soon as the list of extensions gets quite long? Like rn it's only 2 cause that's the two most important plaintext filetypes I could think of but the idea is that the list would get extended quite a bit later down the line.

Reply:
how many filetypes will you need?

Reply:
idk

Reply:
Resolving as per https://github.com/R2Northstar/NorthstarLauncher/pull/682#pullrequestreview-2449702477